### DISCUSSION TITLE
Error. Can't upload image

### DISCUSSION THREAD

**Discussion by igorbochkarev** [outline/outline#2522] [Self Hosting]
Hello,
I have installed Ouline from docker-compose configuration - https://github.com/chsasank/outline-wiki-docker-compose
But have unexpected error "Sorry, error occured uploading image" when trying to insert image in page.
![изображение](https://user-images.githubusercontent.com/5811048/131659123-041aa1cb-d016-4779-b230-9b2abfc82830.png)
Docker logs have no any errors.
Please, help me to find the problem.

**Comment by gopireddy1135** [comment]
This is because your local storage minio container not configured properly. It's working for me, make sure to have https url in .env file AWS_S3_UPLOAD_BUCKET_URL=https://yourdomain.com
Follow this documentation for outline setup https://www.blackvoid.club/outline-wiki-for-growing-teams/
For your reference this is my minio container docker compose service, I am using minio as proxy to my azure blob storage
```
minio:
    image: minio/minio
    container_name: azure-s3
    ports:
      - "9000:9000"
      - "9001:9001"
    volumes:
      - /volume1/docker/minio:/data
    environment:
      - MINIO_ROOT_USER=username
      - MINIO_ROOT_PASSWORD=yourpassword
    command: "gateway azure --console-address :9001"
    restart: always
    networks:
      - mynet
``

**Comment by igorbochkarev** [comment]
Hello,
Unfortunately, I still have an error and can't upload image.
My configuration:
- docker-compose.yaml from https://github.com/chsasank/outline-wiki-docker-compose
- AWS_S3_UPLOAD_BUCKET_URL=https://.... in  env.outline

docker log nginx:
```
192.168.111.1 - - [12/Oct/2021:05:19:38 +0000] "POST /api/documents.update HTTP/1.1" 200 705 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:93.0) Gecko/20100101 Firefox/93.0" "192.168.111.2"
192.168.111.1- - [12/Oct/2021:05:19:38 +0000] "POST /api/attachments.create HTTP/1.1" 200 900 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:93.0) Gecko/20100101 Firefox/93.0" "192.168.111.2"
192.168.111.1- - [12/Oct/2021:05:19:38 +0000] "POST /outline-bucket HTTP/1.1" 404 9 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:93.0) Gecko/20100101 Firefox/93.0" "192.168.111.2"
```
I see 404 error and don't understand why it is.

**Comment by scott-dunt** [comment]
I would check your S3 and AWS permissions:

IAM security policy I use for the user ID for S3 access

{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "s3:ListBucket"
            ],
            "Resource": [
                "arn:aws:s3:::<Bucket Name>"
            ]
        },
        {
            "Sid": "VisualEditor",
            "Effect": "Allow",
            "Action": [
                "s3:GetObjectAcl",
                "s3:DeleteObject",
                "s3:PutObject",
                "s3:GetObject",
                "s3:PutObjectAcl"
            ],
            "Resource": "arn:aws:s3:::<Bucket Name>/*"
        }
    ]
}
S3 Buckets in AWS for use with outline need to have a 'CORS' policy, which is NOT the 'bucket security policy', added to them:

[
    {
        "AllowedHeaders": [
            "*"
        ],
        "AllowedMethods": [
            "PUT",
            "POST"
        ],
        "AllowedOrigins": [
            "https://outline.XXXXX.com"
        ],
        "ExposeHeaders": []
    },
    {
        "AllowedHeaders": [],
        "AllowedMethods": [
            "GET"
        ],
        "AllowedOrigins": [
            "*"
        ],
        "ExposeHeaders": []
    }
]

full notes on docker setup on AWS are here: https://github.com/outline/outline/discussions/2383
